lib/src/network/model/notification/notification_model.dart

```diff
@@ -0,0 +1,67 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'dart:convert';
+
+import 'package:myapp/src/network/data/notification/notification_repository_mock.dart';
+
+class NotificationModel {
+  TypeNotify type;
+  dynamic data;
+  NotificationModel({
+    required this.type,
+    required this.data,
+  });
+
+  NotificationModel copyWith({
+    TypeNotify? type,
+    dynamic data,
+  }) {
+    return NotificationModel(
+      type: type ?? this.type,
+      data: data ?? this.data,
+    );
+  }
+
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'type': type.name.toString(),
+      'data': data,
+    };
+  }
+
+  factory NotificationModel.fromMap(Map<String, dynamic> map) {
+    return NotificationModel(
+      type: _getTypeNotifyFromString(map['status']),
+      data: map['data'] as dynamic,
+    );
+  }
+
+  static TypeNotify _getTypeNotifyFromString(String typeNotifyString) {
```

Comment:
Move this method into TypeNotify enum (Use it as an extension, this is a new use for enums in flutter, I mentioned this before)